kubemarine/patches/p3_reconfigure_registries.py

```diff
@@ -25,18 +25,18 @@ def __init__(self) -> None:
 
     def run(self, res: DynamicResources) -> None:
         logger = res.logger()
-        raw_inventory = res.raw_inventory()
         cluster = res.cluster()
-
         if cluster.inventory['services']['cri']['containerRuntime'] == 'docker':
             logger.info("Docker cri is used, updating containerd configuration is not needed")
             return
-        if not raw_inventory.get('registry'):
-            logger.info("No registries configured, updating containerd configuration is not needed")
-            return
-        result, _ = contains_old_format_properties(raw_inventory)
-        if result:
-            logger.info("Found old format configuration, updating containerd configuration is not needed")
+
+        prev_cluster = res.create_deviated_cluster({
+            'p3_reconfigure_registries': False
+        })
+
+        if 'containerdRegistriesConfig' not in cluster.inventory['services']['cri'] or \
+                'containerdRegistriesConfig' in prev_cluster.inventory['services']['cri']:
```

Comment:
Not sure that I understand these conditions. Is the second condition can really be True in some cases? Consider that 'containerdRegistriesConfig' cannot not be specified for old inventories. Maybe the first condition is enough and create_deviated_cluster is not necessary at all. Alternatively, the result of old and new enrichment can be compared by just `cluster.inventory['services']['cri'] == prev_cluster.inventory['services']['cri']`

Reply:
'containerdRegistriesConfig' for `prev_cluster` checks, that patch is not applied after manual changes in cluster.yaml. In that case user should manually run containerd reconfiguration, but patch should do nothing.
In other cases `cluster.inventory['services']['cri'] == prev_cluster.inventory['services']['cri']` handles the same cases, for this reason it doesn't matter what is used in that condition.